doc/requirements/requirements.md

```diff
@@ -0,0 +1,306 @@
+
+![Command Pipeline with Support](./images/Command-Pipeline-Support.png)
+
+To facilitate error recovery, each module in the chain, if it is unable to carry out its task, its able to propagate error information to its predecessor in the command chain.
+
+## 1.5 Design Goals
+
+The Navigation System designers should strive to meet the following high-level design goals:
+
+* **Modularity** - The Navigation Stack should allow for other implementers to easily replace architectural level components with alternative implementations, in particular, the global and local planners.
+* **Portability** - The Navigation Stack should not introduce inherent limitations in the architectural blocks. For example, it should support multiple kinds of robots, not making assumptions about robot capabilities and limitations and should support various map types and orientations.
+* **Performance** - *TODO: What are the performance goals?* 
+* **Scalability** - *TODO: How low should the implementation scale?* *Specify a minimum platform?*
+* *Other important design goals?*
+
+In particular, there are several specific goals for the ROS 2 Navigation System with respect to improving the existing ROS navigation stack:
+
+* The global and local planners should be easy to replace and chain together in a dynamic way. For example, a user may want to have a global planner for classic point-A-to-point-B travel, but upon reaching point B, have a special planner then control a series of maneuvers such as docking to a charging station or a conveyor belt. 
+* The user should be able to plan complex missions which include a series of movements, and the Navigation System should be able to accept those mission plans and execute them. 
+* The user should be able to specify different types of Robot drive types, such as Ackerman (automobile) steering, and Robot shapes (the current navigation stack is very limited in these areas).
+* The Navigation System should be able to handle more map types and orientations, including outdoor and 2D+ terrain.
+
+# 2.0 Requirements
+
+This section lists the requirements for the Navigation System.
+
+## 2.1 Implementation Constraints
+
+There are various constraints on the development of the ROS 2 Navigation stack.
+
+Id | Handle | Priority | Description | Notes
+-- | ------ | -------- | ----------- | -----
+IC001 | Developer's Guide | 1 | The Navigation System SHALL be developed in accordance with the ROS 2 Devloper's Guide | [ROS 2 Developer's Guide](https://github.com/ros2/ros2/wiki/Developer-Guide)
+IC002 | Implementation Language.C++.Version | 1 | Developers SHALL assume the availability of C++14 language features | Per the ROS 2 Developer's Guide
```

Comment:
Also, should be a SHOULD.

We MUST build on Windows, where the Visual Studio compiler almost certainly doesn't support all C++14 features.

Reply:
Hmm. The ROS2 Developer's Guide mentions C++14 and I just called it out here for emphasis. I suggest we do assume C++14 and deal with any issues encountered with the Microsoft compiler (conditionalize, if necessary).

Reply:
If there's a known list of unsupported C++14 features in Windows, let's review that offline. Do we know if such a list exists?